src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        MachineProducer machineProducer = null;
+
+        if (type.equals(Bulldozer.class)) {
+            machineProducer = new BulldozerProducer();
+        }
+        if (type.equals(Truck.class)) {
+            machineProducer = new TruckProducer();
+        }
+        if (type.equals(Excavator.class)) {
+            machineProducer = new ExcavatorProducer();
+        }
```

Comment:
lets make producers a class-level fields - to avoid creating a new instance each time this method is invoked

Reply:
not fixed. you still create producers in the method :)